src/librbd/internal.cc

```diff
@@ -1913,6 +1921,20 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force) {
       unknown_format = false;
       id = ictx->id;
 
+      if (force) {
+        C_SaferCond ctx;
+        ictx->exclusive_lock->shut_down(&ctx);
```

Comment:
If the exclusive lock feature is enabled, the `exclusive_lock` should be non-null -- but if an notify event comes in on the header while you have it open, there is a potential for a race if you aren't holding the lock.  If the exclusive lock feature isn't enabled (e.g. this wasn't a mirrored image, but just a regular ol' ```rbd rm foo```-call, it might be null)